src/SFA.DAS.ApprenticeCommitments/Application/Commands/UpdateApprenticeship/UpdateApprenticeshipCommandHandler.cs

```diff
@@ -66,21 +60,33 @@ public class UpdateApprenticeshipCommandHandler : IRequestHandler<UpdateApprenti
                 CommitmentsApprovedOn = command.CommitmentsApprovedOn,
             });
 
-            return Unit.Value;
+            return default;
         }
 
-        private async Task<(TrainingProviderResponse, ApprenticeshipResponse apprenticeship, StandardApiResponse)>
+        private async Task<(ApprenticeshipResponse, TrainingProviderResponse, StandardApiResponse)?>
             GetExternalData(UpdateApprenticeshipCommand command)
         {
             var apprenticeship = await _commitmentsService.GetApprenticeshipDetails(
                 command.CommitmentsApprenticeshipId);
 
+            if (IsNullOrEmpty(apprenticeship.Email))
+            {
+                _logger.LogInformation("Apprenticeship {apprenticeshipId} does not have an email, no point in calling Apprentice Commitments", apprenticeship.Id);
+                return default;
+            }
+
+            if (apprenticeship.CourseCode.Contains("-"))
+            {
+                _logger.LogInformation("Apprenticeship {apprenticeshipId} is for a framework, no point in calling Apprentice Commitments", apprenticeship.Id);
+                return default;
+            }
```

Comment:
We could log this second one as a Warning as it should never happen. It would mean we have put an email against an old apprenticeship record. But I'll leave that up to you.

Reply:
Good idea, I've done that 👍